### ISSUE TITLE
question about graphcam

### ISSUE THREAD

**Issue by byte-dance** [vkola-lab/tmi2022#9]
I read your source code, and have a question about the graphcam. 
The code "for index_ in range(3):",I wonder whether the number 3 is fixed or determined by the classes in the classification, for example, for a 5-class task, whether the code should be changed to for index_ in range(5):? 

if graphcam_flag:
            print('GraphCAM enabled')
            p = F.softmax(out)
            torch.save(p, 'graphcam/prob.pt')
            index = np.argmax(out.cpu().data.numpy(), axis=-1)

            **for index_ in range(3):**
                one_hot = np.zeros((1, out.size()[-1]), dtype=np.float32)
                one_hot[0, index_] = out[0][index_]
                one_hot_vector = one_hot
                one_hot = torch.from_numpy(one_hot).requires_grad_(True)
                one_hot = torch.sum(one_hot.cuda() * out)       #!!!!!!!!!!!!!!!!!!!!out-->p
                self.transformer.zero_grad()
                one_hot.backward(retain_graph=True)

                kwargs = {"alpha": 1}
                cam = self.transformer.relprop(torch.tensor(one_hot_vector).to(X.device), method="transformer_attribution", is_ablation=False, 
                                            start_layer=0, **kwargs)

                torch.save(cam, 'graphcam/cam_{}.pt'.format(index_))

        return pred,labels,loss

**Comment by liqiang1102** [comment]
Hi did you figure out graphcam, may I know where you find this part code in the folder? 
Your reply is appreciated.

**Comment by CarlinLiao** [comment]
Agreed. I think that instead of 3 it should be `num_classes = p.size(1)`. If it's more convenient for @GSWS or @vkola-lab I could put in a PR to adjust this, as if it isn't changed I'll likely end up forking it for my own use.

https://github.com/vkola-lab/tmi2022/blob/6d72bfa44c40bd29289d555ffc1f77ffa7f4ce8e/models/GraphTransformer.py#L79-L98

**Comment by GSWS** [comment]
Pull request approved. Thank you for your valuable contribution.
